### PR TITLE
Support preferredScreenCaptureFormat in test options

### PR DESCRIPTION
**Pull request by tatagrigory**
### Note
- Based on the XcodeProj PR https://github.com/tuist/XcodeProj/pull/792
- Will require to update the XcodeProj dependencies in `Package.swift` and `Tuist/Dependencies.swift` when the new version of XcodeProj is released.

### Short description 📝

- Xcode 15 supports automatic screen recordings for UI tests.
- From [Xcode 15 beta Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-15-release-notes#New-Features-in-Xcode-15-Beta):
> XCTest now supports automatic screen recordings in addition to screenshots. Screen recordings allow for a fine-grained view of what happened during a test run, which can be meaningful to investigate UI test failures. Screen recordings are enabled by default (in favor of screenshots). They can be disabled in the test plan or the scheme’s Test action options.
- Available options are:
  - Screenshots - prior to Xcode 15 is was the only available format
  - Screen Recording - new feature of Xcode 15
- It can be configured in the scheme test action options
   <details><summary> New option screenshot </summary>
   Screenshot 2023-09-01 at 17 21 35
   </details>

### How to test the changes locally 🧐

There are three paths:
- Explicit Test Action option
  Example:
  - Ensure `preferredScreenCaptureFormat` is set to `.screenshots` in line 38 of `projects/tuist/fixtures/ios_app_with_custom_scheme/Workspace.swift`
  - Run `swift run tuist generate --path projects/tuist/fixtures/ios_app_with_custom_scheme`
  - Open the generated workspace with Xcode 15
  - Open "Workspace-App" scheme editor
  - Ensure test options have `Automatically capture` set to `screenshots` 
  - Open the generated workspace with Xcode 14
  - Ensure test options are not affected, `Capture screenshots automatically` is selected
  - Repeat test with `preferredScreenCaptureFormat: .screenRecording`
- Project autogenerated scheme options
  Example:
  - Ensure `preferredScreenCaptureFormat`  is set to `.screenshots` in line 10 of `projects/tuist/fixtures/ios_app_with_tests/Project.swift`
  - Run `swift run tuist generate --path projects/tuist/fixtures/ios_app_with_tests `
  - Open the generated project with Xcode 15
  - Open "AppTests" scheme editor
  - Ensure test options have `Automatically capture` set to `screenshots` 
  - Check other scheme, containing tests: "AppUITests", "AppCoreTests", "MacFrameworkTests"
  - Open the generated workspace with Xcode 14
  - Ensure test options are not affected, `Capture screenshots automatically` is selected
  - Repeat test with `testScreenCaptureFormat: .screenRecording`
- Workspace autogenerated scheme options
  Example:
  - Add `, testScreenCaptureFormat: .screenshots` to line 9 of `projects/tuist/fixtures/workspace_with_test_locale/Workspace.swift`
  - Run `swift run tuist generate --path projects/tuist/fixtures/workspace_with_test_locale`
  - Open the generated workspace with Xcode 15
  - Open "App-Workspace" scheme editor
  - Ensure test options have `Automatically capture` set to `screenshots` 
  - Open the generated workspace with Xcode 14
  - Ensure test options are not affected, `Capture screenshots automatically` is selected
  - Repeat test with `testScreenCaptureFormat: .screenRecording`

### Contributor checklist ✅

- [x] The code has been linted using run `./fourier lint tuist --fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Tests/TuistGeneratorTests/Generator/SchemeDescriptorsGeneratorTests.swift [hunk 1]
```diff
@@ -715,6 +715,53 @@ final class SchemeDescriptorsGeneratorTests: XCTestCase {
...
+            scheme: scheme,
+            graphTraverser: graphTraverser,
+            rootPath: try AbsolutePath(validating: "/somepath/Workspace"),
+            generatedProjects: createGeneratedProjects(projects: [project])
+        )
+
+        // Then
+        let result = try XCTUnwrap(got)
+        XCTAssertEqual(result.selectedDebuggerIdentifier, "Xcode.DebuggerFoundation.Debugger.LLDB")
+        XCTAssertEqual(result.selectedLauncherIdentifier, "Xcode.DebuggerFoundation.Launcher.LLDB")
```
Comment (kwridan) [inline Tests/TuistGeneratorTests/Generator/SchemeDescriptorsGeneratorTests.swift:761]: For this test verifying those properties isn't necessary (it's not the purpose of this test)

File: Package.swift [hunk 2]
```diff
@@ -63,7 +63,7 @@ let package = Package(
         .package(url: "https://github.com/SwiftDocOrg/GraphViz.git", exact: "0.2.0"),
         .package(url: "https://github.com/SwiftGen/StencilSwiftKit.git", exact: "2.10.1"),
         .package(url: "https://github.com/SwiftGen/SwiftGen", exact: "6.5.1"),
-        .package(url: "https://github.com/tuist/XcodeProj.git", exact: "8.14.0"),
+        .package(url: "https://github.com/tatagrigory/XcodeProj", branch: "tg/add-preferredScreenCaptureFormat"),
```
Comment (kwridan) [inline Package.swift:66]: For reference: https://github.com/tuist/XcodeProj/pull/792

**Review by kwridan — COMMENTED** [review]
Nice work @tatagrigory 👍 

Worth updating one of the fixtures with to leverage the new option as a show case / demo (e.g. `ios_app_with_custom_scheme`)

We'll have to wait for https://github.com/tuist/XcodeProj/pull/792 to get merged a new release cut before we can proceed with this

**Comment by kwridan** [comment]
@all-contributors add @tatagrigory for code

**Review by kwridan — APPROVED** [review]
Thanks for the updates @tatagrigory 

this is good to go once https://github.com/tuist/XcodeProj/pull/792 is released

File: Package.swift [hunk 3]
```diff
@@ -63,7 +63,7 @@ let package = Package(
         .package(url: "https://github.com/SwiftDocOrg/GraphViz.git", exact: "0.2.0"),
         .package(url: "https://github.com/SwiftGen/StencilSwiftKit.git", exact: "2.10.1"),
         .package(url: "https://github.com/SwiftGen/SwiftGen", exact: "6.5.1"),
-        .package(url: "https://github.com/tuist/XcodeProj.git", exact: "8.14.0"),
```
Comment (pepicrft) [inline Package.swift:66]: @tatagrigory could you update this one to point to 8.15.0. Your changes should be there.

**Review by pepicrft — APPROVED** [review]

**Review by danyf90 — APPROVED** [review]

**Comment by pepicrft** [comment]
Solved some git conflicts. If CI passes I'll go ahead with the merge.

**Pull request merged**